assignment5/unknown_language.py

```diff
@@ -0,0 +1,106 @@
+class UnknownLanguage(object):
+    """UnknownLanguage class contains 3 instance variables,
+    being words_in_lex_order, dictionary and alphabets.
+
+    words_in_lex_order is given as a variable at initialization time
+    and is used to construct the dictionary and the list of alphabets.
+    """
+
+    def __init__(self, words_in_lex_order):
+        self.words_in_lex_order = words_in_lex_order
+        self.alphabets = self._find_alphabets()
+        self.dictionary = self._build_dict()
+
+    def _find_alphabets(self):
+        return set([letter for word in self.words_in_lex_order
```

Comment:
shorter alternative for the alphabet would be to make one long string and to take a set of it:) 
`set(''.join(self.words_in_lex_order))`